src/NodaTime/LocalDate.cs

```diff
@@ -581,6 +581,24 @@ int IComparable.CompareTo(object obj)
         public bool Equals(LocalDate other) => this == other;
 
         /// <summary>
+        /// Resolves this local date into a <see cref="ZonedDateTime"/> in the given time zone representing the
+        /// start of this date in the given zone.
+        /// </summary>
+        /// <remarks>
+        /// This is a convenience method for calling <see cref="DateTimeZone.AtStartOfDay(LocalDate)"/>.
+        /// </remarks>
+        /// <param name="zone">The time zone to map this local date into</param>
+        /// <exception cref="SkippedTimeException">The entire day was skipped due to a very large time zone transition.
+        /// (This is extremely rare.)</exception>
+        /// <returns>The <see cref="ZonedDateTime"/> representing the earliest time on this date, in the given time zone.</returns>
+        [Pure]
+        public ZonedDateTime InZone([NotNull] DateTimeZone zone)
```

Comment:
I like either `AtStartOfDay` or `AtStartOfDayInZone`.  The later more explicitly clarifies the intent, addressing Malcolm's point about differing it from `AtMidnight`.

I don't like `InZone`, as it's not clear at all what time will be chosen.  At least with `AtStartOfDay`, you still have to pass a time zone as a parameter.